install/helm/openchoreo-control-plane/templates/default-resources/component-types/service.yaml

```diff
@@ -141,7 +141,7 @@ spec:
             - name: gateway-external
               namespace: openchoreo-data-plane
           hostnames:
-            - ${environment.name}.${environment.vhost}
+            - ${metadata.environmentName}.${dataplane.publicVirtualHost}
```

Comment:
Doesn't this in build dataplane.publicVirtualHost confuse with the dataplane spec we have? In dataplane's spec we have `Dataplane.Spec.GatewaySpec.PublicVirtualHost`? 

I mean shouldn't we have `publicVirtualHost` in the flat level under dataplane's spec if we are referring like this?

Reply:
This is ok for the moment, I think we need to do a proper cleanup for the fields in the DataPlane CR to make this consistent.